# Materials and Methods

## Data Sources
Accession codes for raw sequencing data originate from Jiao and Schneeberger (2020)[^1]. More information is available [here](https://www.nature.com/articles/s41467-020-14779-y#data-availability). For each accession (An-1, C24, Cvi-0, Ler, and Sha), the analysed data consisted of three sequencing datasets: whole genome Illumina (paired-end short reads), whole genome PacBio (long reads) and whole transcriptome Illumina RNA-seq (paired-end short reads).

## Pre-assembly quality control and statistics
The quality of the obtained fastq files for accession An-1 was evaluated using the FastQC tool (version 0.11.9) [^2] and SeqKit (version 0.13.2) [^3] for additional simple statistics. Optimal k-mer size was estimated twice with Merqury [^4], utilizing the two *A. thaliana* genome sizes reported by Wang et al. (2022)[^5]. The used bash script can be found online [here](https://github.com/marbl/merqury/blob/master/best_k.sh). K-mer counting was performed with Jellyfish (version 2.3.0) [^6] and genome size, composition, and sequencing coverage/quality was assessed with the computed 19-mer count histogram in GenomeScope (online web version available [here](http://qb.cshl.edu/genomescope/)) [^7] for Illumina DNA WGS. To ensure consistent representation of k-mers from paired-end DNA strands, canonical k-mers were employed. By utilizing canonical k-mers, more accurate results with reduced redundancy can be achieved, as the reverse complement of each k-mer is considered equivalent, irrespective of the read orientation in paired-end sequencing.

## Genome and transcriptome assembly
De novo whole genome assembly for PacBio reads was performed using two different assembly tools: Flye (version 2.8.3) [^8] and Canu (version 2.1.1) [^9]. Whole transcriptome assembly was conducted with Trinity (version 2.5.1) [^10].

## Polishing and evaluation
The resulting draft assemblies were polished by aligning the Illumina short reads to the Flye and Canu assemblies using Bowtie 2 (version 2.3.4.1) [^11], using SAMtools (version 1.10) [^12] to sort, index and convert the output SAM files to BAM files, and then utilizing Pilon (version 1.22) [^13] to detect inconsistencies between the assembled genomes and the evidence in the short reads.
The quality of the assembled genomes was then evaluated with the help of three tools. BUSCO (version 4.1.4) [^14] was run on polished and unpolished Flye and Canu assemblies, respectively. The lineage-specific dataset used was brassicales_odb10. Augustus config version was 3.3.3.1 [^15]. BUSCO results were visualized with Python 3.6.13.final.0. The used script can be found [here](https://gitlab.com/ezlab/busco/-/blob/5.5.0/scripts/generate_plot.py)
QUality Assessment Tool for genome assemblies QUAST (version 4.6.0) [^16] was run for polished and unpolished Flye and Canu assemblies with and without reference genome. TAIR10 was used as *A. thaliana* reference genome, specifically Arabidopsis_thaliana.TAIR10.dna.toplevel.fa, which may be found on Ensembl Genomes [^17] [here](https://ftp.ensemblgenomes.ebi.ac.uk/pub/plants/release-57/fasta/arabidopsis_thaliana/dna/).
Lastly, the quality and completeness of the polished and unpolished Flye and Canu assemblies was estimated without the need of a high-quality reference genome, utilizing Merqury (version 1.3-1) [^4].

## Genome comparisons
The polished assembled genomes from Flye and Canu were compared against the *Arabidopsis thaliana* reference using NUCmer, integrated in MUMmer (version 4.0.0beta1) [^18].

## Annotation of Transposable Elements (TEs)
EDTA (version 1.9.6) [^19] [^20] [^21] [^22] [^23] [^24] [^25] [^26] [^27] was employed for the creation of a filtered non-redundant transposable elements (TEs) library for the classification of TEs into the superfamily level as well as whole-genome TE annotation of the polished Flye genome.
The sequences of the two main Class I TEs (or retrotransposons, RTs) superfamilies of interest, Copia and Gypsy, were extracted from the obtained EDTA library - $genome.mod.EDTA.TElib.fa, where $ denotes the file prefix, in this case the assembly name - with SeqKit (version 0.13.2) [^3]. These sequences were then used separately as input for TEsorter (version 1.3.0) [^27] to classify TEs based on protein homology with rexdb-plant as the reference database of transposable element protein domains [^28]. TEsorter analysis was additionally conducted using the Brassicaceae TE database as input file (Brassicaceae_repbase_all_march2019.fasta).

## Dating of TEs / Estimation of insertion age
The raw alignment outputs from RepeatMasker (obtained with EDTA and found under $genome.mod.EDTA.anno/$genome.mod.out) were parsed with the Perl script parseRM.pl (version 5.8.2) [^29] to calculate the divergence. Script was run within a conda environment (version 4.10.1) with Python 3.6.13.final.0 and perl-bioperl 1.7.8. parseRM output was then utilized to date the divergence and to generate a landscape graph for TE superfamilies in R (version 3.6.1, 2019-07-05). The divergence of the annotated TEs was dated to its consensus using the synonymous substitution rate of 8.22 × 10−9 substitutions/synonymous site/year for Brassicaceae species [^30].

## Phylogenetic analysis of TEs
Gypsy RT protein sequences (Ty3-RT) and Copia RT protein sequences (Ty1-RT) were extracted using SeqKit (version 0.13.2) [^3] from the $.rexdb-plant.dom.faa files obtained through TEsorter, where $ denotes either Gypsy or Copia. Extracted sequences were then aligned with clustal omega (version 1.2.4) [^31] and phylogenetic trees created with FastTree (version 2.1.10) [^32]. Trees were annotated and visualized with iTOL (version 6) [^33].

## Annotation of protein-coding sequences
The MAKER pipeline (version 3.01.03) [^34] was used for homology-based genome annotation of protein coding genes through the ab initio prediction of gene structures, alignment based identification of protein-coding genes, and assignment of putative functions to these genes from RNA-Seq data (transcriptome obtained through Trinity) and known protein genes from databases (Atal_v10_CDS_proteins, uniprot-plant_reviewed.fasta).
Annotation completeness of MAKER-predicted proteins was assessed with BUSCO (version 4.1.4) [^14]. Furthermore, annotated proteins were aligned with BLAST+ (version 2.10.1) [^35] against validated proteins with known functions found in the UniProt database (uniprot-plant_reviewed.fasta) to assess sequence homology. Homology was defined as sequence percentage identity of greater than or equal to 98%.

## Comparative genomics
GENESPACE (version 1.1.4) [^36] and OrthoFinder (version v2.5.5) [^37] were used for the identification of syntenic regions and orthogroups. To simplify the visualization and interpretation of the data, only the genes of the 10 longest contigs of each Arabidopsis accession (An-1, C24, Cvi-0, Ler, and Sha) were analysed and compared to the reference genome of *A. thaliana* (TAIR10).

[^1]: Jiao WB, Schneeberger K. Chromosome-level assemblies of multiple Arabidopsis genomes reveal hotspots of rearrangements with altered evolutionary dynamics. Nat Commun. 2020;11(1):989. Published 2020 Feb 20. doi:10.1038/s41467-020-14779-y
[^2]: Andrews S. (2010). FastQC: a quality control tool for high throughput sequence data. Available online at: https://www.bioinformatics.babraham.ac.uk/projects/fastqc/.
[^3]: Shen W, Le S, Li Y, Hu F. SeqKit: A Cross-Platform and Ultrafast Toolkit for FASTA/Q File Manipulation. PLoS One. 2016;11(10):e0163962. Published 2016 Oct 5. doi:10.1371/journal.pone.0163962
[^4]: Rhie A, Walenz BP, Koren S, Phillippy AM. Merqury: reference-free quality, completeness, and phasing assessment for genome assemblies. Genome Biol. 2020;21(1):245. Published 2020 Sep 14. doi:10.1186/s13059-020-02134-9
[^5]: Wang B, Yang X, Jia Y, et al. High-quality Arabidopsis thaliana Genome Assembly with Nanopore and HiFi Long Reads. Genomics Proteomics Bioinformatics. 2022;20(1):4-13. doi:10.1016/j.gpb.2021.08.003
[^6]: Marçais G, Kingsford C. A fast, lock-free approach for efficient parallel counting of occurrences of k-mers. Bioinformatics. 2011;27(6):764-770. doi:10.1093/bioinformatics/btr011
[^7]: Vurture GW, Sedlazeck FJ, Nattestad M, et al. GenomeScope: fast reference-free genome profiling from short reads. Bioinformatics. 2017;33(14):2202-2204. doi:10.1093/bioinformatics/btx153
[^8]:	Kolmogorov M, Yuan J, Lin Y, Pevzner PA. Assembly of long, error-prone reads using repeat graphs. Nat Biotechnol. 2019;37(5):540-546. doi:10.1038/s41587-019-0072-8
[^9]: Koren S, Walenz BP, Berlin K, Miller JR, Bergman NH, Phillippy AM. Canu: scalable and accurate long-read assembly via adaptive k-mer weighting and repeat separation. Genome Res. 2017;27(5):722-736. doi:10.1101/gr.215087.116
[^10]: Grabherr MG, Haas BJ, Yassour M, et al. Full-length transcriptome assembly from RNA-Seq data without a reference genome. Nat Biotechnol. 2011;29(7):644-652. Published 2011 May 15. doi:10.1038/nbt.1883
[^11]: Langmead B, Salzberg SL. Fast gapped-read alignment with Bowtie 2. Nat Methods. 2012;9(4):357-359. Published 2012 Mar 4. doi:10.1038/nmeth.1923
[^12]: Danecek P, Bonfield JK, Liddle J, et al. Twelve years of SAMtools and BCFtools. Gigascience. 2021;10(2):giab008. doi:10.1093/gigascience/giab008. Available from: https://www.htslib.org/. Used version documentation can be found under https://www.htslib.org/doc/1.10/samtools.html
[^13]: Walker BJ, Abeel T, Shea T, et al. Pilon: an integrated tool for comprehensive microbial variant detection and genome assembly improvement. PLoS One. 2014;9(11):e112963. Published 2014 Nov 19. doi:10.1371/journal.pone.0112963
[^14]: Manni M, Berkeley MR, Seppey M, Simão FA, Zdobnov EM. BUSCO Update: Novel and Streamlined Workflows along with Broader and Deeper Phylogenetic Coverage for Scoring of Eukaryotic, Prokaryotic, and Viral Genomes. Mol Biol Evol. 2021;38(10):4647-4654. doi:10.1093/molbev/msab199
[^15]: Stanke M, Diekhans M, Baertsch R, Haussler D. Using native and syntenically mapped cDNA alignments to improve de novo gene finding. Bioinformatics. 2008;24(5):637-644. doi:10.1093/bioinformatics/btn013. Available online at: https://github.com/Gaius-Augustus/Augustus
[^16]: Mikheenko A, Valin G, Prjibelski A, Saveliev V, Gurevich A. Icarus: visualizer for de novo assembly evaluation. Bioinformatics. 2016;32(21):3321-3323. doi:10.1093/bioinformatics/btw379
[^17]: Yates AD, Allen J, Amode RM, et al. Ensembl Genomes 2022: an expanding genome resource for non-vertebrates. Nucleic Acids Res. 2022;50(D1):D996-D1003. doi:10.1093/nar/gkab1007
[^18]: Marçais G, Delcher AL, Phillippy AM, Coston R, Salzberg SL, Zimin A. MUMmer4: A fast and versatile genome alignment system. PLoS Comput Biol. 2018;14(1):e1005944. Published 2018 Jan 26. doi:10.1371/journal.pcbi.1005944
[^19]: Ou S, Su W, Liao Y, et al. Benchmarking transposable element annotation methods for creation of a streamlined, comprehensive pipeline. Genome Biol. 2019;20(1):275. Published 2019 Dec 16. doi:10.1186/s13059-019-1905-y. Available online at: https://github.com/oushujun/EDTA
[^20]: Ellinghaus D, Kurtz S, Willhoeft U. LTRharvest, an efficient and flexible software for de novo detection of LTR retrotransposons. BMC Bioinformatics. 2008;9:18. Published 2008 Jan 14. doi:10.1186/1471-2105-9-18
[^21]: Xu Z, Wang H. LTR_FINDER: an efficient tool for the prediction of full-length LTR retrotransposons. Nucleic Acids Res. 2007;35(Web Server issue):W265-W268. doi:10.1093/nar/gkm286
[^22]: Ou S, Jiang N. LTR_FINDER_parallel: parallelization of LTR_FINDER enabling rapid identification of long terminal repeat retrotransposons. Mob DNA. 2019;10:48. Published 2019 Dec 12. doi:10.1186/s13100-019-0193-0
[^23]: Ou S, Jiang N. LTR_retriever: A Highly Accurate and Sensitive Program for Identification of Long Terminal Repeat Retrotransposons. Plant Physiol. 2018;176(2):1410-1422. doi:10.1104/pp.17.01310
[^24]: Su W, Gu X, Peterson T. TIR-Learner, a New Ensemble Method for TIR Transposable Element Annotation, Provides Evidence for Abundant New Transposable Elements in the Maize Genome. Mol Plant. 2019;12(3):447-460. doi:10.1016/j.molp.2019.02.008
[^25]: Shi J, Liang C. Generic Repeat Finder: A High-Sensitivity Tool for Genome-Wide De Novo Repeat Detection. Plant Physiol. 2019;180(4):1803-1815. doi:10.1104/pp.19.00386
[^26]: Xiong W, He L, Lai J, Dooner HK, Du C. HelitronScanner uncovers a large overlooked cache of Helitron transposons in many plant genomes. Proc Natl Acad Sci U S A. 2014;111(28):10263-10268. doi:10.1073/pnas.1410068111
[^27]: Zhang RG, Li GY, Wang XL, et al. TEsorter: an accurate and fast method to classify LTR-retrotransposons in plant genomes. Hortic Res. Published online February 19, 2022. doi:10.1093/hr/uhac017. Available online at: https://github.com/zhangrengang/TEsorter
[^28]: Neumann P, Novák P, Hoštáková N, Macas J. Systematic survey of plant LTR-retrotransposons elucidates phylogenetic relationships of their polyprotein domains and provides a reference for element classification. Mob DNA. 2019;10:1. Published 2019 Jan 3. doi:10.1186/s13100-018-0144-1
[^29]: Kapusta A, Suh A, Feschotte C. Dynamics of genome size evolution in birds and mammals. Proc Natl Acad Sci U S A. 2017;114(8):E1460-E1469. doi:10.1073/pnas.1616702114. parseRM.pl script is available online at: https://github.com/4ureliek/Parsing-RepeatMasker-Outputs
[^30]: Kagale S, Robinson SJ, Nixon J, et al. Polyploid evolution of the Brassicaceae during the Cenozoic era. Plant Cell. 2014;26(7):2777-2791. doi:10.1105/tpc.114.126391
[^31]: Sievers F, Wilm A, Dineen D, et al. Fast, scalable generation of high-quality protein multiple sequence alignments using Clustal Omega. Mol Syst Biol. 2011;7:539. Published 2011 Oct 11. doi:10.1038/msb.2011.75. Available online at: http://www.clustal.org/omega/
[^32]: Price MN, Dehal PS, Arkin AP. FastTree 2--approximately maximum-likelihood trees for large alignments. PLoS One. 2010;5(3):e9490. Published 2010 Mar 10. doi:10.1371/journal.pone.0009490. Available online at: http://www.microbesonline.org/fasttree/
[^33]: Letunic I, Bork P. Interactive Tree Of Life (iTOL) v5: an online tool for phylogenetic tree display and annotation. Nucleic Acids Res. 2021;49(W1):W293-W296. doi:10.1093/nar/gkab301. Available online at: https://itol.embl.de/
[^34]: Cantarel BL, Korf I, Robb SM, et al. MAKER: an easy-to-use annotation pipeline designed for emerging model organism genomes. Genome Res. 2008;18(1):188-196. doi:10.1101/gr.6743907. Available online from: https://yandell-lab.org/software/maker.html
[^35]: Camacho C, Coulouris G, Avagyan V, et al. BLAST+: architecture and applications. BMC Bioinformatics. 2009;10:421. Published 2009 Dec 15. doi:10.1186/1471-2105-10-421. Available online at: https://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/2.10.1/
[^36]: Lovell JT, Sreedasyam A, Schranz ME, et al. GENESPACE tracks regions of interest and gene copy number variation across multiple genomes. Elife. 2022;11:e78526. Published 2022 Sep 9. doi:10.7554/eLife.78526. Available online at: https://github.com/jtlovell/GENESPACE
[^37]: Emms DM, Kelly S. OrthoFinder: phylogenetic orthology inference for comparative genomics. Genome Biol. 2019;20(1):238. Published 2019 Nov 14. doi:10.1186/s13059-019-1832-y. Available online at: https://github.com/davidemms/OrthoFinder
